pkg/labels/consul_aggregator.go

```diff
@@ -54,16 +71,38 @@ type consulAggregator struct {
 	labeledCache    []Labeled // cached contents of the label subtree
 	aggregatorQuit  chan struct{}
 	aggregationRate time.Duration
+
+	metReg MetricsRegistry
+	// how many watchers are currently using this aggregator?
+	metWatchCount metrics.Gauge
+	// count how many watcher channels are full when a send is attempted
+	metWatchSendMiss metrics.Gauge
+	// how big is the cache of labels?
+	metCacheSize metrics.Gauge
 }
 
-func NewConsulAggregator(labelType Type, kv consulutil.ConsulLister, logger logging.Logger) *consulAggregator {
+func NewConsulAggregator(labelType Type, kv consulutil.ConsulLister, logger logging.Logger, metReg MetricsRegistry) *consulAggregator {
+	if metReg == nil {
+		metReg = metrics.NewRegistry()
+	}
+	watchCount := metrics.NewGauge()
+	watchSendMiss := metrics.NewGauge()
+	cacheSize := metrics.NewGauge()
+	metReg.Register(fmt.Sprintf("%v_aggregate_watches", labelType.String()), watchCount)
+	metReg.Register(fmt.Sprintf("%v_aggregate_send_miss", labelType.String()), watchSendMiss)
+	metReg.Register(fmt.Sprintf("%v_aggregate_cache_size", labelType.String()), cacheSize)
```

Comment:
intentionally not checking `err` on these? the only err i think there could be is a duplicate metric which should be fine